src/Cli/Microsoft.TemplateEngine.Cli/TemplatePackageCoordinator.cs

```diff
@@ -207,9 +207,14 @@ internal async Task<NewCommandStatus> EnterInstallFlowAsync(InstallCommandArgs a
 
             foreach (string installArg in args.TemplatePackages)
             {
-                string[] splitByColons = installArg.Split(new[] { "::" }, StringSplitOptions.RemoveEmptyEntries);
-                string identifier = splitByColons[0];
-                string? version = splitByColons.Length > 1 ? splitByColons[1] : null;
+                string[] splitByColons = installArg.Split(["::"], StringSplitOptions.RemoveEmptyEntries);
+                string[] splitByAt = installArg.Split('@', StringSplitOptions.RemoveEmptyEntries);
+                string[] split = splitByColons.Length > splitByAt.Length ? splitByColons :
+                                 splitByAt.Length > splitByColons.Length ? splitByAt :
```

Comment:
Why is this monstrosity necessary?

I wanted to avoid breaking changes. If one array is longer, that means it split more, and that means we should use it. If the arrays are the same length but the thing at the first index is shorter in one than the other, that means it parsed out (and dropped) (a) separator character(s), and we should use the shorter one.

We should really change CanInstallRemoteNuGetPackage_LatestVariations when we can finally get rid of support for ::

Reply:
Are you ever supposed to have more than one instance of either `::` or `@` in the argument?

It seems like you could check if the string contains `::`.  If it does, split on that, otherwise try splitting it on `@`.

Reply:
Hmm...there could be multiple...which also makes me wonder whether anyone would try `dotnet new install package1::version1 package2@version2`. The semi-reasonable reason someone might do that is if they have some script that takes optional 'additional' package@version things and runs dotnet new install with some 'base' package::version things.

What would you think of something like:
string[] split = installArgs.Split(["::"], StringSplitOptions.RemoveEntryEntries).Select(arg => arg.Split('@', StringSplitOptions.RemoveEmptyEntries)).Flatten().ToArray();

? That should support both cases.